components/compliance-service/integration_test/reporting_server_list_suggestions_test.go

```diff
@@ -0,0 +1,206 @@
+package integration_test
+
+import (
+	"context"
+	"fmt"
+	"os"
+	"testing"
+	"time"
+
+	"github.com/golang/protobuf/jsonpb"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+
+	iam_v2 "github.com/chef/automate/api/interservice/authz/v2"
+	authzConstants "github.com/chef/automate/components/authz-service/constants/v2"
+	"github.com/chef/automate/components/compliance-service/api/reporting"
+	reportingServer "github.com/chef/automate/components/compliance-service/api/reporting/server"
+	"github.com/chef/automate/components/compliance-service/ingest/events/compliance"
+	"github.com/chef/automate/components/compliance-service/reporting/relaxting"
+	rules_tags "github.com/chef/automate/lib/authz"
+)
+
+func ingestReport(fileName string, f func(*compliance.Report)) error {
+	fileData, err := os.Open(fileName)
+	if err != nil {
+		return err
+	}
+	defer fileData.Close()
+	var iReport compliance.Report
+	unmarshaler := &jsonpb.Unmarshaler{AllowUnknownFields: true}
+	if err = unmarshaler.Unmarshal(fileData, &iReport); err != nil {
+		return err
+	}
+
+	f(&iReport)
+
+	ctx := context.Background()
+	_, err = suite.ComplianceIngestServer.ProcessComplianceReport(ctx, &iReport)
+
+	return err
+}
+
+func waitFor(f func() bool) {
+	period := time.Millisecond * 10
+
+	for {
+		if f() {
+			break
+		}
+
+		time.Sleep(period)
+	}
+}
+
+func TestReportingListSuggestions(t *testing.T) {
+	reportFileName := "../ingest/examples/compliance-success-tiny-report.json"
+	everythingCtx := contextWithProjects([]string{authzConstants.AllProjectsExternalID})
+
+	server := reportingServer.New(&relaxting.ES2Backend{ESUrl: elasticsearchUrl})
+
+	n := 5
+
+	reportIds := make([]string, n)
+
+	for i := 0; i < n; i++ {
+		err := ingestReport(reportFileName, func(r *compliance.Report) {
+			id := newUUID()
+
+			r.Environment = id
+			r.NodeName = id
+			r.Platform.Name = id
+			r.Profiles[0].Controls = r.Profiles[0].Controls[:1]
+			r.Profiles[0].Controls[0].Id = id
+			r.Profiles[0].Controls[0].Title = id
+			r.Profiles = r.Profiles[:1]
+			r.Profiles[0].Sha256 = id
+			r.Profiles[0].Title = id
+			r.Recipes = []string{id}
+			r.ReportUuid = id
+			r.Roles = []string{id}
```

Comment:
This is probably the grossest part of this test. Because this test is testing suggestions, I need to do two things:
* Make each report have a unique value in the fields that we suggest on so that they don't get rolled up into one,
* Make it so that the suggestions returned can be tied back to the reports they came from.